### PR TITLE
Change plans page compare plans button label

### PR DESCRIPTION
**Pull request by agrullon95**
## Proposed Changes

* Update `Compare plans` button label to `Compare all plans` for the Guided onboarding flow:
(**The button label should not change for other flows**)

CleanShot 2024-06-12 at 11 57 54@2x

## Why are these changes being made?

* pfGA22-1B-p2#comment-139

## Testing Instructions

* Checkout branch or use live link
* Navigate to `/start/guided` and go through the flow until you reach the plans page.
* Guided onboarding flow plan page: Confirm the `Compare all plans` label appears correctly and the button functions as expected.
* Navigate to `/start/` and go through the flow until you reach the plans page.
* The plan comparison button label should display `Compare plans`

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/my-sites/plans-features-main/index.tsx [hunk 1]
```diff
@@ -679,6 +679,18 @@ const PlansFeaturesMain = ( {
 		? getWooExpressFeaturesGroupedForFeaturesGrid()
 		: getPlanFeaturesGroupedForFeaturesGrid();
 
+	const getComparisonGridToggleLabel = () => {
+		if ( showPlansComparisonGrid ) {
+			return translate( 'Hide comparison' );
+		}
+		switch ( flowName ) {
+			case ONBOARDING_GUIDED_FLOW:
```
Comment (alshakero) [inline client/my-sites/plans-features-main/index.tsx]: For a more general solution, can we use `if ( gridPlansForFeaturesGrid.length < gridPlansForComparisonGrid.length ) ` instead?

**Review by alshakero — CHANGES_REQUESTED** [review]

File: client/my-sites/plans-features-main/index.tsx (hunk 1 above)
Comment (agrullon95) [reply client/my-sites/plans-features-main/index.tsx]: @alshakero - Can you elaborate a bit on this? Whats the purpose of the `gridPlansForFeaturesGrid.length < gridPlansForComparisonGrid.length` comparison?

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/14553738

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @agrullon95 for including a screenshot in the description! This is really helpful for our translators.

File: client/my-sites/plans-features-main/index.tsx (hunk 1 above)
Comment (alshakero) [reply client/my-sites/plans-features-main/index.tsx]: The button needs to say "Compare all plans" when the PlansForFeaturesGrid has fewer plans than the PlansForComparisonGrid. As in, the features grid is showing **_some_** plans, do you want **_all_** plans? 

This translates to `gridPlansForFeaturesGrid.length < gridPlansForComparisonGrid.length`

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/14553738).

**Pull request merged**